lib/commons/dom/focus-disabled.js

```diff
@@ -14,13 +14,49 @@ function focusDisabled(el) {
     return true;
   }
 
+  // if a form element is in a legend, that element will not be disabled even if the fieldset is
+  // https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset
+  let parentNode = vNode.parent;
+  const ancestors = [];
+  let fieldsetDisabled = false;
+  while (parentNode && !fieldsetDisabled) {
```

Comment:
If we know we're not in the same shadow tree anymore we can stop looking at the parent tree completely as well. This also removes the need to check for this in the if statements as well.

```suggestion
  while (parentNode && parentNode.shadowId === vNode.shadowId && !fieldsetDisabled) {
```

Reply:
Also, we'll want to break once we know whether or not there is a parent fieldset disabled. Right now this only breaks if the value is true, but we want to short circuit the entire tree as soon as any node has the cached value regardless of what the value is.